LayoutTests/http/tests/paymentrequest/paymentrequest-merchantCategoryCode.https.html

```diff
@@ -0,0 +1,73 @@
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    setup({ explicit_done: true, explicit_timeout: true });
+    test((test) => {
+        const method = validPaymentMethod();
+        method.data.features = ["paymentRequestMerchantCategoryCode"];
+        new PaymentRequest([method], validPaymentDetails());
+    }, "Should have a feature for `paymentRequestMerchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const request = new PaymentRequest([validPaymentMethod()], validPaymentDetails());
+        
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        
+        const shippingPromise = new Promise(resolve => {
+            request.addEventListener("shippingaddresschange", (event) => {
+                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after an update");
+                event.updateWith({ });
+                internals.mockPaymentCoordinator.acceptPayment();
+                
+                resolve();
+            });
+        });
+        
+        const response = await request.show();
+        
+        await shippingPromise;
+        
+        assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after the payment is accepted");
+        
+        await response.complete("success");
+    }, "Should not have a default value for `merchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const method = validPaymentMethod();
+        method.data.merchantCategoryCode = "0739";
+        
+        const request = new PaymentRequest([method], validPaymentDetails());
+        
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        
+        const shippingPromise = new Promise(resolve => {
+            request.addEventListener("shippingaddresschange", (event) => {
+                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, method.data.merchantCategoryCode, "check that the `merchantCategoryCode` still matches after an update");
+                event.updateWith({ });
+                internals.mockPaymentCoordinator.acceptPayment();
+                
+                resolve();
+            });
+        });
+        
+        const response = await request.show();
+        
+        await shippingPromise;
+        
+        assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, method.data.merchantCategoryCode, "check that the `merchantCategoryCode` still matches after the payment is accepted");
+        
+        await response.complete("success");
+    }, "Should propagate `merchantCategoryCode` if provided.");
+    
+    done();
```

Comment:
Is this done() needed?

Reply:
Yeah, try removing it. I don't think it does anything.

Reply:
If it's removed the tests time out